rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton-v3.c

```diff
@@ -0,0 +1,93 @@
+#include <stdio.h>
+#include <unistd.h>
+#include <sys/wait.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <linux/types.h>
+#include <sys/stat.h>
+#include "liberpal-skeleton.h"
+#include "sgx_call.h"
+#include "defines.h"
+
+int pal_get_version(void)
+{
+	return 3;
+}
+
+int pal_init(pal_attr_t *attr)
+{
+	return __pal_init(attr);
+}
+
+int pal_create_process(pal_create_process_args *args)
+{
+	if (args == NULL || args->path == NULL || args->argv == NULL || args->pid == NULL || args->stdio == NULL) {
+		errno = EINVAL;
+		return -1;
+        }
+
+	int pid;
+	if (!is_oot_driver) {
+		if ((pid = fork()) < 0)
+			return -1;
+		else if (pid == 0) {
+			int exit_code, ret;
+
+			ret = __pal_exec(args->path, args->argv, args->stdio, &exit_code);
+			exit(ret ? ret : exit_code);
+		} else
+			*args->pid = pid;
+	}
+
+	return 0;
+}
+
+int pal_exec(pal_exec_args *attr)
+{
+	if (attr == NULL || attr->exit_value == NULL) {
+		errno = EINVAL;
+		return -1;
+	}
+
+	int status;
+	if (!is_oot_driver) {
+		waitpid(attr->pid, &status, 0);
+
+		if (WIFEXITED(status) || WIFSIGNALED(status))
+			*attr->exit_value = WEXITSTATUS(status);
+	} else {
+		while(1) {
+			printf("Hello Skeleton!\n");
+			printf("    - Powered by ACK-TEE and runE\n");
+			fflush(stdout);
+			sleep(3);
+		}
```

Comment:
Don't do this. Just call __pal_exec().

Reply:
In addition, I think you need to write a long comments to explain why pal_create_process and pal_exec need to check oot driver or in-tree driver.